src/services/blockchain/__tests__/eth-bc-service.test.ts

```diff
@@ -88,3 +86,185 @@ describe('ETH service',  () => {
+      blockNumber: 54321,
+      transactionHash: txnResponse.hash,
+    }
+    const block = {
+      timestamp: 54321000
+    }
+
+    provider.getGasPrice.mockReturnValue(gasPrice)
+    provider.estimateGas.mockReturnValue(gasEstimate)
+    wallet.sendTransaction.mockReturnValue(txnResponse)
+    provider.waitForTransaction.mockReturnValue(txReceipt)
+    provider.getBlock.mockReturnValue(block)
+
+    const txRequest = {
+      to: wallet.address,
+      data: "0x987654321",
+      nonce,
+      gasPrice: gasPrice,
+      gasLimit: gasEstimate,
+    }
+    const tx = await ethBc._trySendTransaction(txRequest, 0, "eip155:1337", 10)
+    expect(tx).toMatchSnapshot()
+
+    const txData = wallet.sendTransaction.mock.calls[0][0]
+    expect(txData).toMatchSnapshot()
+  });
+
+  test('successful mocked transaction', async () => {
+    const balance = 10 * 1000 * 1000
+    const nonce = 5
+    const gasPrice = BigNumber.from(1000)
+    const gasEstimate = BigNumber.from(10*1000)
+
+    provider.getBalance.mockReturnValue(balance)
+    provider.getTransactionCount.mockReturnValue(nonce)
+    provider.getGasPrice.mockReturnValue(gasPrice)
+    provider.estimateGas.mockReturnValue(gasEstimate)
+
+    const mockTrySendTransaction = jest.fn()
+    ethBc._trySendTransaction = mockTrySendTransaction
+    mockTrySendTransaction.mockReturnValue({txHash: "0x12345"})
+
+    const cid = new CID('bafyreic5p7grucmzx363ayxgoywb6d4qf5zjxgbqjixpkokbf5jtmdj5ni');
+    await ethBc.sendTransaction(cid)
+
+    expect(mockTrySendTransaction).toHaveBeenCalledTimes(1)
+    const [txData, attemptNum, network, transactionTimeoutSecs] = mockTrySendTransaction.mock.calls[0]
+    expect(attemptNum).toEqual(0)
+    expect(network).toEqual(config.blockchain.connectors.ethereum.network)
+    expect(transactionTimeoutSecs).toEqual(config.blockchain.connectors.ethereum.transactionTimeoutSecs)
+    expect(txData).toMatchSnapshot()
+  });
+
+  test('insufficient funds error', async () => {
+    const balance = BigNumber.from(10 * 1000 * 1000)
+    const nonce = 5
+    const gasPrice = BigNumber.from(1000)
+    const gasEstimate = BigNumber.from(10*1000)
+
+    provider.getBalance.mockReturnValue(balance)
+    provider.getTransactionCount.mockReturnValue(nonce)
+    provider.getGasPrice.mockReturnValue(gasPrice)
+    provider.estimateGas.mockReturnValue(gasEstimate)
+
+    const mockTrySendTransaction = jest.fn()
+    ethBc._trySendTransaction = mockTrySendTransaction
+    mockTrySendTransaction.mockRejectedValue({code: ErrorCode.INSUFFICIENT_FUNDS})
+
+    const cid = new CID('bafyreic5p7grucmzx363ayxgoywb6d4qf5zjxgbqjixpkokbf5jtmdj5ni');
+    await expect(ethBc.sendTransaction(cid)).rejects.toThrow(/Transaction cost is greater than our current balance/)
+
+    // In first attempt gas cost is exactly equal to wallet balance, in second attempt it goes
+    // over the wallet balance and the whole attempt is aborted
+    expect(mockTrySendTransaction).toHaveBeenCalledTimes(2)
```

Comment:
Yeah, it was confusing as it was written since we were forcing the mock wallet to return INSUFFICIENT_FUNDS when we actually did in fact have sufficient funds as written.  I updated it so the first time it errors with a timeout and it's only the second time that it errors with insufficient funds, which matches better what we're likely to see in the real world anyway